File: leetcode/easy/804. Unique Morse Code Words/solution.cc
#include<string>
#include<vector>
#include<set>

using namespace std;

vector<string> table = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};

class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        set<string> s;
        for (string word: words) {
            string str = "";
            for (char ch: word) {
                str += table[(int)(ch) - 97];
            }
            s.insert(str);
        }
        return s.size();
    }
};

File: leetcode/easy/804. Unique Morse Code Words/solution.js
// https://leetcode.com/problems/unique-morse-code-words/
/**
 * @param {string[]} words
 * @return {number}
 */
function uniqueMorseRepresentations(words) {
    const table = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."];
    const set = new Set();
    for (const word of words) {
        let str = '';
        for (const char of word) {
            str += table[char.charCodeAt(0) - 97];
        }
        set.add(str);
    }
    return set.size;
};

uniqueMorseRepresentations(["gin", "zen", "gig", "msg"])